astrapy/db.py

```diff
@@ -76,6 +76,45 @@ def _post(self, path=None, document=None):
         )
         return response
 
+    def _pre_process_find(self, vector, fields=None):
+        # Must pass a vector
+        if not vector:
+            return ValueError("Must pass a vector")
+
+        # Build the new vector parameter
+        sort = {"$vector": vector}
+
+        # Build the new fields parameter
+        projection = {f: 1 for f in fields} if fields else {}
+
+        # TODO: Do we always return the similarity?
+        projection["$similarity"] = 1
+
+        return sort, projection
+
+    def _post_process_find(
+        self,
+        raw_find_result,
+        include_similarity=True,
+        _key="documents",
+    ):
+        # If we only have a single result, treat it as a list
+        if type(raw_find_result["data"][_key]) == dict:
```

Comment:
Faulty API responses would surface to the user as KeyError here, presumably. Worth adding a check for the keys perhaps (and raise a ValueError with the jsondumps of the response) ?

Reply:
So with the new coding logic, I'm going to resolve this for now, but please comment if you think more should be done